### ISSUE TITLE
Produce Dropwizard timer metrics instead of gauges in MetricsFilter

### ISSUE THREAD

**Issue by lidkowiak** [spring-projects/spring-boot#4405]
Here [spring-boot-dropwizard-metric-filter](https://github.com/lidkowiak/spring-boot-dropwizard-metric-filter) I implemented [DropwizardMetricsFilter](https://github.com/lidkowiak/spring-boot-dropwizard-metric-filter/blob/master/src/main/java/pl/lidkowiak/DropwizardMetricsFilter.java) which produces timer metrics instead of only last response time. It is based on MetricsFilter and adds e.g. percentiles and rates metrics.

What do you think about merging it into Spring Boot Actuator? It could be enabled via AutoConfiguration when Dropwizard Metrics is on the classpath and/or with ConditionalOnProperty.

**Comment by golonzovsky** [comment]
Are there any plans to proceed on this?  

Maybe having Timer service in MetricsFilter (which could be enabled for dropwizard) would be a less duplicating approach.  Seems there are some related requests #4196

Or the idea behind MetricsFilter is to limit recorded types only to ones which are displayed under /metrics endpoint? It might be least intrusive change to have some extension point under [MetricsFilter#submitMetrics](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/MetricsFilter.java#L186), so filter code could be reused for other cases.

With current implementation without code change it seems that recording of timers can be done by [wrapping gauge service](http://stackoverflow.com/questions/38794197/whats-the-optimal-way-to-connect-spring-boot-with-statsd-especially-for-timers)

Is this issue related/blocked by #4909 and/or #2949 ?

**Comment by golonzovsky** [comment]
In case someone will need this - managed to report gauges as timers with current implementation with following override: 

```java
    @Bean
    DropwizardMetricServices dropwizardMetricServices(MetricRegistry metricRegistry,
                                                      ObjectProvider<ReservoirFactory> resFactoryProvider) {
        return new DropwizardMetricServices(metricRegistry, resFactoryProvider.getIfAvailable()) {
            @Override
            public void submit(String name, double value) {
                if (name.startsWith("response.")) {
                    super.submit("timer." + name, value);
                } else {
                    super.submit(name, value);
                }
            }
        };
    }
```

Resulting timers: 
```
-- Timers ----------------------------------------------------------------------
timer.response.half
             count = 59
         mean rate = 1.12 calls/second
     1-minute rate = 1.62 calls/second
     5-minute rate = 2.36 calls/second
    15-minute rate = 2.52 calls/second
               min = 11.00 milliseconds
               max = 498.00 milliseconds
              mean = 214.34 milliseconds
            stddev = 137.68 milliseconds
            median = 225.00 milliseconds
              75% <= 299.00 milliseconds
              95% <= 438.00 milliseconds
              98% <= 463.00 milliseconds
              99% <= 498.00 milliseconds
            99.9% <= 498.00 milliseconds
```

**Comment by wilkinsona** [comment]
This has been superseded by the planned move to [Micrometer](http://micrometer.io)-based metrics (#9970)
